### ISSUE TITLE
Android location question - office move

### ISSUE THREAD

**Issue by chuckf** [owntracks/android#966]
* **App build number**:  23024
* **Android version**: 9   
* **Device**: Samsung S8

This is not directly an Owntracks problem, but is related to how it gets location. I hope its not a problem posting such a question.

Recently I've had an office move. Its about 45 minutes away from the previous location. When I arrive at the new location, Owntracks shows the proper location via GPS. However, once I enter the building the phone uses the wifi (I'm assuming) to determine the address. This shows the previous address of the office since its the previously used gear. I've found some solutions on the web that boils down to 'it takes time for Google to learn the new location'. Is there a 'reset wifi location' type of function available that I'm just not finding?

**Comment by growse** [comment]
Yeah, you're running into a common problem of any service that tries to locate you based on the WiFi BSSID. I had exactly the same thing when I moved house - my location would 'bounce' between the old and new house (even though they were tens of miles apart) as the Google fused location thingie used various different sources to discover where I was.

From an OT perspective, there's not much to be done at the moment. The device is assumed to give an accurate location. That said, a non-google flavour is being worked on which should use just the native android location service - this won't use Google's WiFi database to provide a fix. It may give slightly less reliable results, but at least you won't get the bouncing.

I think some people have had some success swapping out the Google provider for another service like Mozilla's using microg, but I'm not sure how supported that is, and can't recommend it (I've not tried it recently).

In theory, the app itself should be able to support other providers like MLS, but I think this might be a bit of work to implement, plus Mozilla aren't currently handing out API keys....

In my case, the way I got it fixed was repeatedly opening Google maps whenever I saw the location was wrong and doing a "blue dot is wrong" report. Frustrating, but it's the only way I know of to get the Google database fixed quickly.
